readthedocs/doc_builder/python_environments.py

```diff
@@ -121,6 +122,29 @@ def install_package(self, install):
                 bin_path=self.venv_bin(),
             )
 
+    def _pip_cache_cmd_argument(self):
+        """
+        Return the pip command ``--cache-dir`` or ``--no-cache-dir`` argument.
+
+        The decision is made considering if the directories are going to be
+        cleaned after the build (``RTD_CLEAN_AFTER_BUILD=True`` or project has
+        the ``CLEAN_AFTER_BUILD`` feature enabled). In this case, there is no
+        need to cache anything.
+        """
+        if (
+            not settings.RTD_CLEAN_AFTER_BUILD and
+            not self.project.has_feature(Feature.CLEAN_AFTER_BUILD)
```

Comment:
This should be more simple if the conditional is inverted.

```suggestion
            settings.RTD_CLEAN_AFTER_BUILD or
            self.project.has_feature(Feature.CLEAN_AFTER_BUILD)
        ):
            return ['--no-cache-dir'] 
```

Github suggestion still don't work on multiple lines, but you get it.

Reply:
I copied the logic from the task itself (https://github.com/readthedocs/readthedocs.org/blob/30ede06ea1b9bb85ab618bc0e656be2cf7e6f237/readthedocs/projects/tasks.py#L1507-L1515). I didn't want to invert it so it's easier to map with the other.